CMAKE options must be optional (Openmp is not manatory).

-- The C compiler identification is AppleClang 9.0.0.9000039
-- The CXX compiler identification is AppleClang 9.0.0.9000039
-- Check for working C compiler: /Library/Developer/CommandLineTools/usr/bin/cc
-- Check for working C compiler: /Library/Developer/CommandLineTools/usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /Library/Developer/CommandLineTools/usr/bin/c++
-- Check for working CXX compiler: /Library/Developer/CommandLineTools/usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Error at /usr/local/Cellar/cmake/3.10.2/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:137 (message):
  Could NOT find OpenMP_C (missing: OpenMP_C_FLAGS OpenMP_C_LIB_NAMES)
Call Stack (most recent call first):
  /usr/local/Cellar/cmake/3.10.2/share/cmake/Modules/FindPackageHandleStandardArgs.cmake:378 (_FPHSA_FAILURE_MESSAGE)
  /usr/local/Cellar/cmake/3.10.2/share/cmake/Modules/FindOpenMP.cmake:447 (find_package_handle_standard_args)
  examples/CMakeLists.txt:45 (find_package)

--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -24,9 +24,21 @@ target_link_libraries ("laik"
 # Optional MPI backend
 if (mpi-backend)
     find_pkgconfig ("mpi")
-    find_program (mpirun "mpirun")
 
-    if (TARGET "mpi" AND mpirun)
+    # If pkg-config didn't find MPI above, also try CMake's native FindMPI
+    # module since we really want to enable MPI support whenever possible!
+    if (NOT TARGET "mpi")
+        find_package (MPI)
+        if (MPI_C_FOUND)
+            add_library ("mpi" INTERFACE)
+            target_include_directories ("mpi" INTERFACE "${MPI_C_INCLUDE_PATH}")
+            target_compile_options     ("mpi" INTERFACE "${MPI_C_COMPILE_FLAGS}")
+            target_link_libraries      ("mpi" INTERFACE "${MPI_C_LINK_FLAGS}")
+            target_link_libraries      ("mpi" INTERFACE "${MPI_C_LIBRARIES}")
+        endif ()
+    endif ()
+
+    if (TARGET "mpi")
         target_sources ("laik"
             PRIVATE "backend-mpi.c"
         )
@@ -43,7 +55,6 @@ if (mpi-backend)
     else ()
         message (FATAL_ERROR "Dependency check for option 'mpi-backend' failed, stopping!")
     endif ()
-
 endif ()
 
 # Optional Single backend

